src/components/RightHandMenu/DonutChart/Donut.js

```diff
@@ -1,34 +1,119 @@
+
+//   const myCalc = (midAngle, sin, sy) => {
+//     if (
+//       midAngle <= 10 ||
+//       midAngle >= 350 ||
+//       (midAngle >= 170 && midAngle <= 190)
+//     ) {
+//       return sy
+//     } else if (midAngle >= 340 || (midAngle >= 180 && midAngle <= 220)) {
+//       return cy + (outerRadius + 30) * sin + 20
+//     } else if (midAngle <= 20 || (midAngle >= 140 && midAngle <= 180)) {
+//       return cy + (outerRadius + 30) * sin - 20
+//     } else {
+//       return cy + (outerRadius + 30) * sin
+//     }
+//   }
+
+//   const sin = Math.sin(-RADIAN * midAngle)  //convert degree to radian units and find sin value of it
+//   const cos = Math.cos(-RADIAN * midAngle)  //convert degree to radian units and find cos value of it
+//   const sx = cx + outerRadius * cos         //start coordinates (sx,sy) for line
+//   const sy = cy + outerRadius * sin
+//   const mx = midAngle <= 10 || midAngle >= 350 || (midAngle >= 170 && midAngle <= 190) //mid coordinates (mx,my) for line
+//               ? cx + (outerRadius + 30) * cos
+//               : sx
+//   const my = myCalc(midAngle, sin, sy)
+//   const ex = midAngle <= 10 || midAngle >= 350 || (midAngle >= 170 && midAngle <= 190)  //end coordinates (ex,ey) for line
+//               ? mx
+//               : mx + (cos >= 0 ? 1 : -1) * 22   //1st and 4th quadrant cos is +ve ; 1st and 2nd quadrant sin is +ve
+//   const ey = my
+//   const textAnchor = cos >= 0 ? 'end' : 'start' //where to anchor text tag
+//   const tx= ex + (cos >= 0 ? 1 : -1) * 12       //coordinates (tx,ty) on which text tag is placed
+//   const ty= ey + (sin >= 0 ? 1 : -1) * 16
+
+//   return (
+//     <g>
+//       {/* centers text in middle of pie chart */}
+//       {/* <text x={cx} y={cy} dy={8} textAnchor="middle" fill={fill}>
+//         {payload.name}
+//         </text> */}
+
+//       {/* line from pie chart to label name */}
+//       <path
+//         d={`M${sx},${sy}L${mx},${my}L${ex},${ey}`}
+//         stroke={fill}
+//         fill='none'
+//       />
+
+//       {/* Circle at the tip of line */}
+//       <circle cx={ex} cy={ey} r={3} fill={fill} stroke='none' />
+
+//       {/* Label name/ text at end tip of line from pie chart */}
+//       <text
+//         textLength='30'
+//         x={tx}
+//         y={ty}
+//         textAnchor={textAnchor}
+//       >
+//         <tspan 
+//           fill='#1c752a'
+//           x={tx}
+//           y={ty} >{`${payload.payload.key}`}</tspan>
+//         <tspan
+//           className='tspan__val'
+//           x={tx + (cos >= 0 ? -1 : 1) * 6}
+//           y={ty + 10}>{payload.payload.percent ? `${value} (${payload.payload.percent}%)` : value}</tspan>
+//       </text>
+//     </g>
+//   )
+// }
+
+// Attempt 2: labels on hover
+const renderLabelonHover = (props) => {
   const RADIAN = Math.PI / 180
   const {
     cx,
     cy,
+    startAngle,
     midAngle,
+    endAngle,
+    innerRadius,
     outerRadius,
     fill,
     value,
-    name
+    payload
```

Comment:
Is payload a required prop from recharts? If so, it's good. Otherwise, we might want to name it something more descriptive.